plugin.rb

```diff
@@ -120,41 +118,28 @@ def self.accept_answer!(post, acting_user, topic: nil)
     def self.unaccept_answer!(post, topic: nil)
       topic ||= post.topic
       topic ||= Topic.unscoped.find_by(id: post.topic_id)
-
       return if topic.nil?
+      return if topic.solved.nil?
 
       DistributedMutex.synchronize("discourse_solved_toggle_answer_#{topic.id}") do
-        post.custom_fields.delete(IS_ACCEPTED_ANSWER_CUSTOM_FIELD)
-        topic.custom_fields.delete(ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD)
-
-        if timer_id = topic.custom_fields[AUTO_CLOSE_TOPIC_TIMER_CUSTOM_FIELD]
-          topic_timer = TopicTimer.find_by(id: timer_id)
-          topic_timer.destroy! if topic_timer
-          topic.custom_fields.delete(AUTO_CLOSE_TOPIC_TIMER_CUSTOM_FIELD)
-        end
-
-        topic.save!
-        post.save!
+        solved = topic.solved
+        solved.topic_timer.destroy! if solved.topic_timer
```

Comment:
Not related to this PR but the database operations here should happen within the same transaction.